charts/openstack-cluster/templates/_helpers.tpl

```diff
@@ -178,40 +219,59 @@ files:
     permissions: "0644"
 {{- end }}
 {{- end }}
-{{- if $ctx.Values.registryAuth }}
+{{- if .Values.registryAuth }}
   - path: /etc/containerd/conf.d/auth.toml
     contentFrom:
       secret:
-        name: {{ include "openstack-cluster.componentName" (list $ctx "containerd-auth") }}
+        name: {{ include "openstack-cluster.componentName" (list . "containerd-auth") }}
         key: "auth.toml"
     owner: root:root
     permissions: "0644"
 {{- end }}
-{{- if $trustedCAs }}
-  {{- range $name, $certificate := $trustedCAs }}
+{{- end }}
+
+{{/*
+Produces the kubeadmConfigSpec required to configure additional trusted CAs for cluster nodes,
+e.g. for private registries.
+*/}}
+{{- define "openstack-cluster.kubeadmConfigSpec.trustedCAs" -}}
+{{- with .Values.trustedCAs }}
+files:
+  {{- range $name, $certificate := . }}
   - path: /usr/local/share/ca-certificates/{{ $name }}.crt
     content: |
       {{- nindent 6 $certificate }}
     owner: root:root
     permissions: "0644"
   {{- end }}
+preKubeadmCommands:
+  - update-ca-certificates
 {{- end }}
-{{- if $files }}
-  {{- range $files }}
-  - {{ toYaml . | nindent 4 }}
-  {{- end }}
 {{- end }}
-{{- if or $trustedCAs $additionalPackages $preKubeadmCommands }}
+
+{{/*
+Produces the kubeadmConfigSpec required to install additional packages.
+*/}}
+{{- define "openstack-cluster.kubeadmConfigSpec.additionalPackages" -}}
+{{- with .Values.additionalPackages }}
 preKubeadmCommands:
-  {{- if $trustedCAs }}
-  - update-ca-certificates
-  {{- end }}
-  {{- if $additionalPackages }}
   - apt update -y
-  - apt install -y {{ join " " $additionalPackages }}
-  {{- end }}
-  {{- range $preKubeadmCommands }}
-  - {{ . }}
-  {{- end }}
+  - apt install -y {{ join " " . }}
```

Comment:
Can/should `apt install -y` be parameterised to support other package management commands (e.g. `dnf install -y`)?

Reply:
I was thinking that can follow in subsequent patches, with `openstack-cluster.kubeadmConfigSpec.additionalPackages.{ubuntu,rocky,...}`.

The layering implemented in this patch makes that easier in the future.